paperqa/agents/search.py

```diff
@@ -81,12 +81,11 @@ def read_from_string(self, data: str | bytes) -> BaseModel | SupportsPickle:
 
 
 class SearchIndex:
-
-    REQUIRED_FIELDS: ClassVar[set[str]] = {"file_location", "body"}
+    REQUIRED_FIELDS: ClassVar[list[str]] = ["file_location", "body"]
```

Comment:
`set` is still a `Sequence`, let's keep this as a `set` to enforce uniqueness

Reply:
I see what you're saying but set isn't a Sequence, it's a Collection, right?

Reply:
I think a `Collection` is a subset of `Sequence`, unless I am missing something

Reply:
Other way around I think: https://docs.python.org/3/library/collections.abc.html#collections-abstract-base-classes